validator-cli/src/ArbToEth/validator.ts

```diff
@@ -0,0 +1,92 @@
+import { VeaInboxArbToEth, VeaOutboxArbToEth } from "@kleros/vea-contracts/typechain-types";
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { ethers } from "ethers";
+import { ArbToEthTransactionHandler } from "./transactionHandler";
+import { getClaim, getClaimResolveState } from "../utils/claim";
+import { defaultEmitter } from "../utils/emitter";
+import { BotEvents } from "../utils/botEvents";
+import { getBlocksAndCheckFinality } from "../utils/arbToEthState";
+
+// https://github.com/prysmaticlabs/prysm/blob/493905ee9e33a64293b66823e69704f012b39627/config/params/mainnet_config.go#L103
+const secondsPerSlotEth = 12;
+
+export async function challengeAndResolveClaim(
```

Comment:
When a function has so many parameters, consider using an object instead of positional arguments. That makes invoking the function way easier to read. It also allows you to mock one of the dependencies (for example, fetchClaim) without having to provide a value for a previous value (emitter)